// sarvamai/dist/esm/api/resources/speechToTextTranslateStreaming/client/Client.mjs
// This file was auto-generated by Fern from our API Definition.
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { normalizeClientOptionsWithAuth } from "../../../../BaseClient.mjs";
import { mergeHeaders, mergeOnlyDefinedHeaders } from "../../../../core/headers.mjs";
import * as core from "../../../../core/index.mjs";
import * as environments from "../../../../environments.mjs";
import { SpeechToTextTranslateStreamingSocket } from "./Socket.mjs";
export class SpeechToTextTranslateStreamingClient {
    constructor(options = {}) {
        this._options = normalizeClientOptionsWithAuth(options);
    }
    connect(args) {
        return __awaiter(this, void 0, void 0, function* () {
            var _a, _b;
            const { model, input_audio_codec: inputAudioCodec, sample_rate: sampleRate, high_vad_sensitivity: highVadSensitivity, vad_signals: vadSignals, flush_signal: flushSignal, headers, debug, reconnectAttempts, } = args;
            const _queryParams = {};
            if (model != null) {
                _queryParams.model = model;
            }
            if (inputAudioCodec != null) {
                _queryParams.input_audio_codec = inputAudioCodec;
            }
            if (sampleRate != null) {
                _queryParams.sample_rate = sampleRate;
            }
            if (highVadSensitivity != null) {
                _queryParams.high_vad_sensitivity = highVadSensitivity;
            }
            if (vadSignals != null) {
                _queryParams.vad_signals = vadSignals;
            }
            if (flushSignal != null) {
                _queryParams.flush_signal = flushSignal;
            }
            const _authRequest = yield this._options.authProvider.getAuthRequest();
            const _headers = mergeHeaders(_authRequest.headers, mergeOnlyDefinedHeaders({ "Api-Subscription-Key": args["Api-Subscription-Key"] }), headers);
            const apiSubscriptionKeyValue = _headers["api-subscription-key"];
            const socket = new core.ReconnectingWebSocket({
                url: core.url.join((_a = (yield core.Supplier.get(this._options.baseUrl))) !== null && _a !== void 0 ? _a : ((_b = (yield core.Supplier.get(this._options.environment))) !== null && _b !== void 0 ? _b : environments.SarvamAIEnvironment.Production).production, "/speech-to-text-translate/ws"),
                protocols: [`api-subscription-key.${apiSubscriptionKeyValue}`],
                queryParameters: _queryParams,
                headers: _headers,
                options: { debug: debug !== null && debug !== void 0 ? debug : false, maxRetries: reconnectAttempts !== null && reconnectAttempts !== void 0 ? reconnectAttempts : 30 },
            });
            return new SpeechToTextTranslateStreamingSocket({ socket });
        });
    }
}
