// sarvamai/dist/cjs/api/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.textToSpeechStreaming = exports.textToSpeech = exports.text = exports.speechToTextTranslateStreaming = exports.speechToTextTranslateJob = exports.speechToTextStreaming = exports.speechToTextJob = exports.speechToText = exports.pronunciationDictionary = exports.documentIntelligence = exports.chat = void 0;
__exportStar(require("./chat/client/requests/index.js"), exports);
exports.chat = __importStar(require("./chat/index.js"));
__exportStar(require("./documentIntelligence/client/requests/index.js"), exports);
exports.documentIntelligence = __importStar(require("./documentIntelligence/index.js"));
__exportStar(require("./pronunciationDictionary/client/requests/index.js"), exports);
exports.pronunciationDictionary = __importStar(require("./pronunciationDictionary/index.js"));
__exportStar(require("./speechToText/client/requests/index.js"), exports);
exports.speechToText = __importStar(require("./speechToText/index.js"));
__exportStar(require("./speechToTextJob/client/requests/index.js"), exports);
exports.speechToTextJob = __importStar(require("./speechToTextJob/index.js"));
exports.speechToTextStreaming = __importStar(require("./speechToTextStreaming/index.js"));
__exportStar(require("./speechToTextStreaming/types/index.js"), exports);
__exportStar(require("./speechToTextTranslateJob/client/requests/index.js"), exports);
exports.speechToTextTranslateJob = __importStar(require("./speechToTextTranslateJob/index.js"));
exports.speechToTextTranslateStreaming = __importStar(require("./speechToTextTranslateStreaming/index.js"));
__exportStar(require("./speechToTextTranslateStreaming/types/index.js"), exports);
__exportStar(require("./text/client/requests/index.js"), exports);
exports.text = __importStar(require("./text/index.js"));
__exportStar(require("./textToSpeech/client/requests/index.js"), exports);
exports.textToSpeech = __importStar(require("./textToSpeech/index.js"));
exports.textToSpeechStreaming = __importStar(require("./textToSpeechStreaming/index.js"));
__exportStar(require("./textToSpeechStreaming/types/index.js"), exports);
