// sarvamai/dist/cjs/Client.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SarvamAIClient = void 0;
const Client_js_1 = require("./api/resources/chat/client/Client.js");
const Client_js_2 = require("./api/resources/documentIntelligence/client/Client.js");
const Client_js_3 = require("./api/resources/pronunciationDictionary/client/Client.js");
const Client_js_4 = require("./api/resources/speechToText/client/Client.js");
const Client_js_5 = require("./api/resources/speechToTextJob/client/Client.js");
const Client_js_6 = require("./api/resources/speechToTextStreaming/client/Client.js");
const Client_js_7 = require("./api/resources/speechToTextTranslateJob/client/Client.js");
const Client_js_8 = require("./api/resources/speechToTextTranslateStreaming/client/Client.js");
const Client_js_9 = require("./api/resources/text/client/Client.js");
const Client_js_10 = require("./api/resources/textToSpeech/client/Client.js");
const Client_js_11 = require("./api/resources/textToSpeechStreaming/client/Client.js");
const BaseClient_js_1 = require("./BaseClient.js");
class SarvamAIClient {
    constructor(options = {}) {
        this._options = (0, BaseClient_js_1.normalizeClientOptionsWithAuth)(options);
    }
    get text() {
        var _a;
        return ((_a = this._text) !== null && _a !== void 0 ? _a : (this._text = new Client_js_9.TextClient(this._options)));
    }
    get speechToText() {
        var _a;
        return ((_a = this._speechToText) !== null && _a !== void 0 ? _a : (this._speechToText = new Client_js_4.SpeechToTextClient(this._options)));
    }
    get textToSpeech() {
        var _a;
        return ((_a = this._textToSpeech) !== null && _a !== void 0 ? _a : (this._textToSpeech = new Client_js_10.TextToSpeechClient(this._options)));
    }
    get pronunciationDictionary() {
        var _a;
        return ((_a = this._pronunciationDictionary) !== null && _a !== void 0 ? _a : (this._pronunciationDictionary = new Client_js_3.PronunciationDictionaryClient(this._options)));
    }
    get chat() {
        var _a;
        return ((_a = this._chat) !== null && _a !== void 0 ? _a : (this._chat = new Client_js_1.ChatClient(this._options)));
    }
    get speechToTextJob() {
        var _a;
        return ((_a = this._speechToTextJob) !== null && _a !== void 0 ? _a : (this._speechToTextJob = new Client_js_5.SpeechToTextJobClient(this._options)));
    }
    get speechToTextTranslateJob() {
        var _a;
        return ((_a = this._speechToTextTranslateJob) !== null && _a !== void 0 ? _a : (this._speechToTextTranslateJob = new Client_js_7.SpeechToTextTranslateJobClient(this._options)));
    }
    get documentIntelligence() {
        var _a;
        return ((_a = this._documentIntelligence) !== null && _a !== void 0 ? _a : (this._documentIntelligence = new Client_js_2.DocumentIntelligenceClient(this._options)));
    }
    get speechToTextStreaming() {
        var _a;
        return ((_a = this._speechToTextStreaming) !== null && _a !== void 0 ? _a : (this._speechToTextStreaming = new Client_js_6.SpeechToTextStreamingClient(this._options)));
    }
    get speechToTextTranslateStreaming() {
        var _a;
        return ((_a = this._speechToTextTranslateStreaming) !== null && _a !== void 0 ? _a : (this._speechToTextTranslateStreaming = new Client_js_8.SpeechToTextTranslateStreamingClient(this._options)));
    }
    get textToSpeechStreaming() {
        var _a;
        return ((_a = this._textToSpeechStreaming) !== null && _a !== void 0 ? _a : (this._textToSpeechStreaming = new Client_js_11.TextToSpeechStreamingClient(this._options)));
    }
}
exports.SarvamAIClient = SarvamAIClient;
