// sarvamai/dist/cjs/play.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.toWavBytes = toWavBytes;
exports.save = save;
exports.play = play;
const index_js_1 = require("./core/runtime/index.js");
/**
 * Concatenates `TextToSpeechResponse.audios` WAV base64 chunks into a single WAV byte array
 * (matching the Python SDK helper).
 *
 * If `options.index` is provided, it decodes only that one entry (no concatenation).
 */
function toWavBytes(audio, options = {}) {
    const audios = Array.isArray(audio) ? audio : audio === null || audio === void 0 ? void 0 : audio.audios;
    if (!Array.isArray(audios)) {
        const keys = audio != null && typeof audio === "object"
            ? Object.keys(audio)
                .slice(0, 20)
                .join(", ")
            : "";
        throw new Error("Invalid audio input: expected `string[]` (base64 WAV chunks) or `{ audios: string[] }` (TextToSpeechResponse). " +
            "If you're using streaming TTS, collect `msg.data.audio` chunks into an array and pass that array to `audio.toWavBytes(...)` / `audio.save(...)`." +
            (keys ? ` Received object keys: ${keys}` : ""));
    }
    if (audios.length === 0) {
        throw new Error("No audio data found (audios is empty).");
    }
    if (options.index != null) {
        const chunk = audios[options.index];
        if (chunk == null) {
            throw new Error(`Invalid audio index ${options.index}.`);
        }
        return base64ToBytes(chunk);
    }
    if (audios.length === 1) {
        return base64ToBytes(audios[0]);
    }
    const segments = [];
    const first = base64ToBytes(audios[0]);
    segments.push(first);
    for (let i = 1; i < audios.length; i++) {
        const chunk = base64ToBytes(audios[i]);
        const dataPos = findAscii(chunk, "data");
        if (dataPos === -1) {
            // If we can't find a data chunk, fall back to appending everything (best effort)
            segments.push(chunk);
            continue;
        }
        const audioDataStart = dataPos + 8; // "data" (4) + chunkSize (4)
        segments.push(chunk.subarray(audioDataStart));
    }
    const combined = concatBytes(segments);
    // Update RIFF chunk size at bytes [4..7]
    // RIFF size = fileSize - 8
    writeUint32LE(combined, 4, combined.length - 8);
    // Update data chunk size at "data"+4
    const dataPos = findAscii(combined, "data");
    if (dataPos !== -1) {
        const dataSize = combined.length - dataPos - 8;
        writeUint32LE(combined, dataPos + 4, dataSize);
    }
    return combined;
}
function save(audio_1, filename_1) {
    return __awaiter(this, arguments, void 0, function* (audio, filename, options = {}) {
        var _a;
        const bytes = toWavBytes(audio, options);
        if (index_js_1.RUNTIME.type === "browser") {
            if (typeof document === "undefined") {
                throw new Error("Browser save requires `document` to trigger a download.");
            }
            // `BlobPart` typings require an ArrayBuffer-backed view; copy to ensure `ArrayBuffer` (not `ArrayBufferLike`).
            const blobBytes = new Uint8Array(bytes);
            const blob = new Blob([blobBytes], { type: "audio/wav" });
            const url = URL.createObjectURL(blob);
            try {
                const a = document.createElement("a");
                a.href = url;
                a.download = filename;
                a.style.display = "none";
                document.body.appendChild(a);
                a.click();
                a.remove();
            }
            finally {
                URL.revokeObjectURL(url);
            }
            return;
        }
        // Node / Bun / Deno best-effort: try `fs` dynamic import (like other parts of this SDK)
        const fs = yield Promise.resolve().then(() => __importStar(require("fs")));
        if (!((_a = fs === null || fs === void 0 ? void 0 : fs.promises) === null || _a === void 0 ? void 0 : _a.writeFile)) {
            throw new Error("Saving to a file is not supported in this environment.");
        }
        yield fs.promises.writeFile(filename, bytes);
    });
}
function play(audio_1) {
    return __awaiter(this, arguments, void 0, function* (audio, options = {}) {
        var _a;
        const bytes = toWavBytes(audio);
        if (index_js_1.RUNTIME.type === "browser") {
            // `BlobPart` typings require an ArrayBuffer-backed view; copy to ensure `ArrayBuffer` (not `ArrayBufferLike`).
            const blobBytes = new Uint8Array(bytes);
            const blob = new Blob([blobBytes], { type: "audio/wav" });
            const url = URL.createObjectURL(blob);
            try {
                const el = new Audio(url);
                // Autoplay behavior depends on browser policies; caller may need a user gesture.
                yield el.play();
            }
            finally {
                // Let playback start; revoke soon after to avoid leaks.
                setTimeout(() => URL.revokeObjectURL(url), 10000);
            }
            return;
        }
        const useFfmpeg = (_a = options.useFfmpeg) !== null && _a !== void 0 ? _a : true;
        if (!useFfmpeg) {
            throw new Error("`useFfmpeg=false` is not supported in Node (no built-in audio output).");
        }
        const childProcess = yield Promise.resolve().then(() => __importStar(require("child_process")));
        yield new Promise((resolve, reject) => {
            var _a, _b, _c;
            const proc = childProcess.spawn("ffplay", ["-autoexit", "-", "-nodisp"], {
                stdio: ["pipe", "ignore", "pipe"],
            });
            proc.on("error", (err) => {
                if ((err === null || err === void 0 ? void 0 : err.code) === "ENOENT") {
                    reject(new Error("ffplay (from ffmpeg) not found, necessary to play audio. " +
                        "On mac you can install it with `brew install ffmpeg`. " +
                        "On linux/windows you can install from https://ffmpeg.org/."));
                }
                else {
                    reject(err);
                }
            });
            (_a = proc.stderr) === null || _a === void 0 ? void 0 : _a.on("data", () => {
                // ignore ffplay logs
            });
            proc.on("close", () => resolve());
            // Write WAV bytes to stdin
            (_b = proc.stdin) === null || _b === void 0 ? void 0 : _b.write(typeof Buffer !== "undefined" ? Buffer.from(bytes) : bytes);
            (_c = proc.stdin) === null || _c === void 0 ? void 0 : _c.end();
        });
    });
}
/** -------- internals -------- */
function base64ToBytes(base64) {
    if (typeof Buffer !== "undefined") {
        const buf = Buffer.from(base64, "base64");
        return new Uint8Array(buf.buffer, buf.byteOffset, buf.byteLength);
    }
    if (typeof atob !== "function") {
        throw new Error("Base64 decoding is not supported in this environment.");
    }
    const bin = atob(base64);
    const out = new Uint8Array(bin.length);
    for (let i = 0; i < bin.length; i++)
        out[i] = bin.charCodeAt(i);
    return out;
}
function concatBytes(chunks) {
    let total = 0;
    for (const c of chunks)
        total += c.length;
    const out = new Uint8Array(total);
    let offset = 0;
    for (const c of chunks) {
        out.set(c, offset);
        offset += c.length;
    }
    return out;
}
function writeUint32LE(buf, offset, value) {
    buf[offset] = value & 0xff;
    buf[offset + 1] = (value >>> 8) & 0xff;
    buf[offset + 2] = (value >>> 16) & 0xff;
    buf[offset + 3] = (value >>> 24) & 0xff;
}
function findAscii(haystack, needle) {
    const n = new Uint8Array(needle.length);
    for (let i = 0; i < needle.length; i++)
        n[i] = needle.charCodeAt(i);
    outer: for (let i = 0; i <= haystack.length - n.length; i++) {
        for (let j = 0; j < n.length; j++) {
            if (haystack[i + j] !== n[j])
                continue outer;
        }
        return i;
    }
    return -1;
}
