// sarvamai/dist/cjs/api/resources/documentIntelligence/DocumentIntelligenceJob.js
"use strict";
// Document Intelligence Job wrapper with convenience methods (matching Python SDK API)
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentIntelligenceJob = void 0;
/**
 * A convenience wrapper around a Document Intelligence job that provides
 * a fluent API matching the Python SDK.
 *
 * @example
 * ```typescript
 * const job = await client.documentIntelligence.createJob({
 *     language: "hi-IN",
 *     outputFormat: "html"
 * });
 *
 * await job.uploadFile("./document.pdf");
 * await job.start();
 * const status = await job.waitUntilComplete();
 * const metrics = job.getPageMetrics();
 * await job.downloadOutput("./output.html");
 * ```
 */
class DocumentIntelligenceJob {
    constructor(client, jobId, options = {}) {
        var _a, _b;
        this._status = null;
        this._client = client;
        this._jobId = jobId;
        this._pollingIntervalMs = (_a = options.pollingIntervalMs) !== null && _a !== void 0 ? _a : 2000;
        this._maxPollingAttempts = (_b = options.maxPollingAttempts) !== null && _b !== void 0 ? _b : 150;
    }
    /** The unique job ID */
    get jobId() {
        return this._jobId;
    }
    /** Alias for jobId (Python SDK compatibility) */
    get job_id() {
        return this._jobId;
    }
    /** The last fetched status (may be null if not yet polled) */
    get status() {
        return this._status;
    }
    /**
     * Upload a file to the job using a presigned URL.
     * Supports file path (Node.js) or File/Blob (browser).
     */
    uploadFile(file) {
        return __awaiter(this, void 0, void 0, function* () {
            // Get filename
            let filename;
            let fileContent;
            if (typeof file === "string") {
                // Node.js: file path
                const fs = yield Promise.resolve().then(() => __importStar(require("fs")));
                const path = yield Promise.resolve().then(() => __importStar(require("path")));
                filename = path.basename(file);
                const buffer = yield fs.promises.readFile(file);
                fileContent = buffer;
            }
            else if (file instanceof File) {
                // Browser: File object
                filename = file.name;
                fileContent = file;
            }
            else {
                // Blob - need a filename
                filename = "document.pdf";
                fileContent = file;
            }
            // Get upload link
            const uploadResponse = yield this._client.getUploadLinks({
                job_id: this._jobId,
                files: [filename],
            });
            const uploadUrls = uploadResponse.upload_urls;
            if (!uploadUrls || Object.keys(uploadUrls).length === 0) {
                throw new Error("No upload URL returned from server");
            }
            // Get the first upload URL
            const uploadInfo = Object.values(uploadUrls)[0];
            if (!(uploadInfo === null || uploadInfo === void 0 ? void 0 : uploadInfo.file_url)) {
                throw new Error("Invalid upload URL response");
            }
            // Upload the file using PUT request
            const headers = {
                "x-ms-blob-type": "BlockBlob",
            };
            // Add any metadata headers from the response
            if (uploadInfo.file_metadata) {
                for (const [key, value] of Object.entries(uploadInfo.file_metadata)) {
                    if (typeof value === "string") {
                        headers[key] = value;
                    }
                }
            }
            const response = yield fetch(uploadInfo.file_url, {
                method: "PUT",
                headers,
                body: fileContent,
            });
            if (!response.ok) {
                throw new Error(`Failed to upload file: ${response.status} ${response.statusText}`);
            }
        });
    }
    /**
     * Start processing the job.
     */
    start() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this._client.start(this._jobId);
            this._status = response;
            return response;
        });
    }
    /**
     * Get the current status of the job.
     */
    getStatus() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this._client.getStatus(this._jobId);
            this._status = response;
            return response;
        });
    }
    /**
     * Poll until the job completes (Completed, PartiallyCompleted, or Failed).
     */
    waitUntilComplete() {
        return __awaiter(this, void 0, void 0, function* () {
            const terminalStates = ["Completed", "PartiallyCompleted", "Failed"];
            let attempts = 0;
            while (attempts < this._maxPollingAttempts) {
                const status = yield this.getStatus();
                if (terminalStates.includes(status.job_state)) {
                    return status;
                }
                yield this._sleep(this._pollingIntervalMs);
                attempts++;
            }
            throw new Error(`Job did not complete within ${this._maxPollingAttempts * this._pollingIntervalMs / 1000} seconds`);
        });
    }
    /**
     * Get page-level metrics from the last status.
     */
    getPageMetrics() {
        var _a, _b, _c, _d, _e, _f;
        if (!this._status) {
            throw new Error("No status available. Call getStatus() or waitUntilComplete() first.");
        }
        const jobDetails = (_a = this._status.job_details) === null || _a === void 0 ? void 0 : _a[0];
        return {
            totalPages: (_b = jobDetails === null || jobDetails === void 0 ? void 0 : jobDetails.total_pages) !== null && _b !== void 0 ? _b : 0,
            pagesProcessed: (_c = jobDetails === null || jobDetails === void 0 ? void 0 : jobDetails.pages_processed) !== null && _c !== void 0 ? _c : 0,
            pagesSucceeded: (_d = jobDetails === null || jobDetails === void 0 ? void 0 : jobDetails.pages_succeeded) !== null && _d !== void 0 ? _d : 0,
            pagesFailed: (_e = jobDetails === null || jobDetails === void 0 ? void 0 : jobDetails.pages_failed) !== null && _e !== void 0 ? _e : 0,
            pageErrors: (_f = jobDetails === null || jobDetails === void 0 ? void 0 : jobDetails.page_errors) !== null && _f !== void 0 ? _f : [],
        };
    }
    /**
     * Download the output file(s) to the specified path.
     * The output is a ZIP file containing the processed documents.
     *
     * @param outputPath - Path where the output file will be saved
     * @returns The path to the downloaded file
     */
    downloadOutput(outputPath) {
        return __awaiter(this, void 0, void 0, function* () {
            const downloadResponse = yield this._client.getDownloadLinks(this._jobId);
            const downloadUrls = downloadResponse.download_urls;
            if (!downloadUrls || Object.keys(downloadUrls).length === 0) {
                throw new Error("No download URLs available. Job may not be complete.");
            }
            // Download the first output file
            const downloadInfo = Object.values(downloadUrls)[0];
            if (!(downloadInfo === null || downloadInfo === void 0 ? void 0 : downloadInfo.file_url)) {
                throw new Error("Invalid download URL response");
            }
            // Fetch with timeout (matching Python SDK's 300s timeout)
            const controller = new AbortController();
            const timeoutId = setTimeout(() => controller.abort(), 300000);
            try {
                const response = yield fetch(downloadInfo.file_url, {
                    signal: controller.signal,
                });
                if (!response.ok) {
                    throw new Error(`Failed to download file: ${response.status} ${response.statusText}`);
                }
                // Use arrayBuffer for binary data (ZIP files)
                const arrayBuffer = yield response.arrayBuffer();
                const content = new Uint8Array(arrayBuffer);
                // Save to file (Node.js) or trigger download (browser)
                if (typeof window === "undefined") {
                    // Node.js
                    const fs = yield Promise.resolve().then(() => __importStar(require("fs")));
                    const path = yield Promise.resolve().then(() => __importStar(require("path")));
                    // Ensure output directory exists (matching Python SDK behavior)
                    const outputDir = path.dirname(outputPath);
                    if (outputDir) {
                        yield fs.promises.mkdir(outputDir, { recursive: true });
                    }
                    yield fs.promises.writeFile(outputPath, content);
                }
                else {
                    // Browser - trigger download
                    const blob = new Blob([content], { type: "application/zip" });
                    const url = URL.createObjectURL(blob);
                    const a = document.createElement("a");
                    a.href = url;
                    a.download = outputPath;
                    document.body.appendChild(a);
                    a.click();
                    a.remove();
                    URL.revokeObjectURL(url);
                }
                return outputPath;
            }
            finally {
                clearTimeout(timeoutId);
            }
        });
    }
    /**
     * Get download links for the output files.
     */
    getDownloadLinks() {
        return __awaiter(this, void 0, void 0, function* () {
            return this._client.getDownloadLinks(this._jobId);
        });
    }
    _sleep(ms) {
        return new Promise((resolve) => setTimeout(resolve, ms));
    }
}
exports.DocumentIntelligenceJob = DocumentIntelligenceJob;
