// sarvamai/dist/esm/api/resources/speechToTextStreaming/types/SpeechToTextStreamingLanguageCode.mjs
// This file was auto-generated by Fern from our API Definition.
/**
 * Specifies the language of the input audio in BCP-47 format.
 *
 * **Available Options (saarika:v2.5, legacy):**
 * - `unknown` (default): Use when the language is not known; the API will auto-detect.
 * - `hi-IN`: Hindi
 * - `bn-IN`: Bengali
 * - `gu-IN`: Gujarati
 * - `kn-IN`: Kannada
 * - `ml-IN`: Malayalam
 * - `mr-IN`: Marathi
 * - `od-IN`: Odia
 * - `pa-IN`: Punjabi
 * - `ta-IN`: Tamil
 * - `te-IN`: Telugu
 * - `en-IN`: English
 *
 * **Additional Options (saaras:v3, recommended):**
 * - `as-IN`: Assamese
 * - `ur-IN`: Urdu
 * - `ne-IN`: Nepali
 * - `kok-IN`: Konkani
 * - `ks-IN`: Kashmiri
 * - `sd-IN`: Sindhi
 * - `sa-IN`: Sanskrit
 * - `sat-IN`: Santali
 * - `mni-IN`: Manipuri
 * - `brx-IN`: Bodo
 * - `mai-IN`: Maithili
 * - `doi-IN`: Dogri
 */
export const SpeechToTextStreamingLanguageCode = {
    Unknown: "unknown",
    EnIn: "en-IN",
    HiIn: "hi-IN",
    BnIn: "bn-IN",
    GuIn: "gu-IN",
    KnIn: "kn-IN",
    MlIn: "ml-IN",
    MrIn: "mr-IN",
    OdIn: "od-IN",
    PaIn: "pa-IN",
    TaIn: "ta-IN",
    TeIn: "te-IN",
    AsIn: "as-IN",
    UrIn: "ur-IN",
    NeIn: "ne-IN",
    KokIn: "kok-IN",
    KsIn: "ks-IN",
    SdIn: "sd-IN",
    SaIn: "sa-IN",
    SatIn: "sat-IN",
    MniIn: "mni-IN",
    BrxIn: "brx-IN",
    MaiIn: "mai-IN",
    DoiIn: "doi-IN",
};
