// sarvamai/dist/cjs/api/types/ConfigureConnection.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigureConnection = void 0;
var ConfigureConnection;
(function (ConfigureConnection) {
    let Data;
    (function (Data) {
        /**
         * Specifies the model to use for text-to-speech conversion.
         * - **bulbul:v2** (default): Standard TTS model with pitch/loudness support
         * - **bulbul:v3**: Advanced model with temperature control (no pitch/loudness)
         */
        Data.Model = {
            BulbulV2: "bulbul:v2",
            BulbulV3: "bulbul:v3",
        };
        /** The language of the text in BCP-47 format */
        Data.TargetLanguageCode = {
            BnIn: "bn-IN",
            EnIn: "en-IN",
            GuIn: "gu-IN",
            HiIn: "hi-IN",
            KnIn: "kn-IN",
            MlIn: "ml-IN",
            MrIn: "mr-IN",
            OdIn: "od-IN",
            PaIn: "pa-IN",
            TaIn: "ta-IN",
            TeIn: "te-IN",
        };
        /**
         * The speaker voice to be used for the output audio.
         *
         * **Model Compatibility:**
         * - **bulbul:v2:** anushka (default), abhilash, manisha, vidya, arya, karun, hitesh
         * - **bulbul:v3:** aditya (default), ritu, priya, neha, rahul, pooja, rohan, simran, kavya, amit, dev, ishita, shreya, ratan, varun, manan, sumit, roopa, kabir, aayan, shubh, ashutosh, advait, amelia, sophia
         *
         * **Note:** Speaker selection must match the chosen model version.
         */
        Data.Speaker = {
            Anushka: "anushka",
            Abhilash: "abhilash",
            Manisha: "manisha",
            Vidya: "vidya",
            Arya: "arya",
            Karun: "karun",
            Hitesh: "hitesh",
            Aditya: "aditya",
            Ritu: "ritu",
            Priya: "priya",
            Neha: "neha",
            Rahul: "rahul",
            Pooja: "pooja",
            Rohan: "rohan",
            Simran: "simran",
            Kavya: "kavya",
            Amit: "amit",
            Dev: "dev",
            Ishita: "ishita",
            Shreya: "shreya",
            Ratan: "ratan",
            Varun: "varun",
            Manan: "manan",
            Sumit: "sumit",
            Roopa: "roopa",
            Kabir: "kabir",
            Aayan: "aayan",
            Shubh: "shubh",
            Ashutosh: "ashutosh",
            Advait: "advait",
            Amelia: "amelia",
            Sophia: "sophia",
        };
        /** Audio codec (currently supports MP3 only, optimized for real-time playback) */
        Data.OutputAudioCodec = {
            Linear16: "linear16",
            Mulaw: "mulaw",
            Alaw: "alaw",
            Opus: "opus",
            Flac: "flac",
            Aac: "aac",
            Wav: "wav",
            Mp3: "mp3",
        };
        /** Audio bitrate (choose from 5 supported bitrate options) */
        Data.OutputAudioBitrate = {
            ThirtyTwoK: "32k",
            SixtyFourK: "64k",
            NinetySixK: "96k",
            OneHundredTwentyEightK: "128k",
            OneHundredNinetyTwoK: "192k",
        };
    })(Data = ConfigureConnection.Data || (ConfigureConnection.Data = {}));
})(ConfigureConnection || (exports.ConfigureConnection = ConfigureConnection = {}));
