// sarvamai/dist/cjs/api/types/SpeechToTextLanguage.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpeechToTextLanguage = void 0;
/**
 * Languages supported for Speech-to-Text.
 *
 * **saarika:v2.5 supports (12 languages):** unknown, hi-IN, bn-IN, kn-IN, ml-IN, mr-IN, od-IN, pa-IN, ta-IN, te-IN, en-IN, gu-IN
 *
 * **saaras:v3 supports all 23 languages** including: as-IN, ur-IN, ne-IN, kok-IN, ks-IN, sd-IN, sa-IN, sat-IN, mni-IN, brx-IN, mai-IN, doi-IN
 */
exports.SpeechToTextLanguage = {
    Unknown: "unknown",
    HiIn: "hi-IN",
    BnIn: "bn-IN",
    KnIn: "kn-IN",
    MlIn: "ml-IN",
    MrIn: "mr-IN",
    OdIn: "od-IN",
    PaIn: "pa-IN",
    TaIn: "ta-IN",
    TeIn: "te-IN",
    EnIn: "en-IN",
    GuIn: "gu-IN",
    AsIn: "as-IN",
    UrIn: "ur-IN",
    NeIn: "ne-IN",
    KokIn: "kok-IN",
    KsIn: "ks-IN",
    SdIn: "sd-IN",
    SaIn: "sa-IN",
    SatIn: "sat-IN",
    MniIn: "mni-IN",
    BrxIn: "brx-IN",
    MaiIn: "mai-IN",
    DoiIn: "doi-IN",
};
