// sarvamai/dist/cjs/api/types/TextToSpeechSpeaker.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TextToSpeechSpeaker = void 0;
exports.TextToSpeechSpeaker = {
    Anushka: "anushka",
    Abhilash: "abhilash",
    Manisha: "manisha",
    Vidya: "vidya",
    Arya: "arya",
    Karun: "karun",
    Hitesh: "hitesh",
    Aditya: "aditya",
    Ritu: "ritu",
    Priya: "priya",
    Neha: "neha",
    Rahul: "rahul",
    Pooja: "pooja",
    Rohan: "rohan",
    Simran: "simran",
    Kavya: "kavya",
    Amit: "amit",
    Dev: "dev",
    Ishita: "ishita",
    Shreya: "shreya",
    Ratan: "ratan",
    Varun: "varun",
    Manan: "manan",
    Sumit: "sumit",
    Roopa: "roopa",
    Kabir: "kabir",
    Aayan: "aayan",
    Shubh: "shubh",
    Ashutosh: "ashutosh",
    Advait: "advait",
    Anand: "anand",
    Tanya: "tanya",
    Tarun: "tarun",
    Sunny: "sunny",
    Mani: "mani",
    Gokul: "gokul",
    Vijay: "vijay",
    Shruti: "shruti",
    Suhani: "suhani",
    Mohit: "mohit",
    Kavitha: "kavitha",
    Rehan: "rehan",
    Soham: "soham",
    Rupali: "rupali",
};
