// sarvamai/dist/esm/api/resources/textToSpeechStreaming/client/Socket.mjs
// This file was auto-generated by Fern from our API Definition.
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import * as core from "../../../../core/index.mjs";
import { fromJson, toJson } from "../../../../core/json.mjs";
export class TextToSpeechStreamingSocket {
    constructor(args) {
        this.eventHandlers = {};
        this.handleOpen = () => {
            var _a, _b;
            (_b = (_a = this.eventHandlers).open) === null || _b === void 0 ? void 0 : _b.call(_a);
        };
        this.handleMessage = (event) => {
            var _a, _b;
            const data = fromJson(event.data);
            (_b = (_a = this.eventHandlers).message) === null || _b === void 0 ? void 0 : _b.call(_a, data);
        };
        this.handleClose = (event) => {
            var _a, _b;
            (_b = (_a = this.eventHandlers).close) === null || _b === void 0 ? void 0 : _b.call(_a, event);
        };
        this.handleError = (event) => {
            var _a, _b;
            const message = event.message;
            (_b = (_a = this.eventHandlers).error) === null || _b === void 0 ? void 0 : _b.call(_a, new Error(message));
        };
        this.socket = args.socket;
        this.socket.addEventListener("open", this.handleOpen);
        this.socket.addEventListener("message", this.handleMessage);
        this.socket.addEventListener("close", this.handleClose);
        this.socket.addEventListener("error", this.handleError);
    }
    /** The current state of the connection; this is one of the readyState constants. */
    get readyState() {
        return this.socket.readyState;
    }
    /**
     * @param event - The event to attach to.
     * @param callback - The callback to run when the event is triggered.
     * Usage:
     * ```typescript
     * this.on('open', () => {
     *     console.log('The websocket is open');
     * });
     * ```
     */
    on(event, callback) {
        this.eventHandlers[event] = callback;
    }
    /**
     * Configure the connection with various options including output audio codec.
     * Accepts both formats:
     * - Flat params: { speaker: "anushka", target_language_code: "hi-IN", ... }
     * - Wire format: { type: "config", data: { speaker: "anushka", target_language_code: "hi-IN", ... } }
     */
    configureConnection(config) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k;
        this.assertSocketIsOpen();
        // Detect if user passed { type: "config", data: {...} } or flat params
        let data;
        if ("type" in config && config.type === "config" && "data" in config) {
            // Wire format: { type: "config", data: {...} }
            data = config.data;
        }
        else {
            // Flat params format
            data = config;
        }
        // Backward compatibility: default to "anushka" but warn if not specified
        const speaker = (_a = data.speaker) !== null && _a !== void 0 ? _a : "anushka";
        const message = {
            type: "config",
            data: {
                target_language_code: data.target_language_code,
                speaker,
                pitch: (_b = data.pitch) !== null && _b !== void 0 ? _b : 0.0,
                pace: (_c = data.pace) !== null && _c !== void 0 ? _c : 1.0,
                loudness: (_d = data.loudness) !== null && _d !== void 0 ? _d : 1.0,
                speech_sample_rate: (_e = data.speech_sample_rate) !== null && _e !== void 0 ? _e : 22050,
                enable_preprocessing: (_f = data.enable_preprocessing) !== null && _f !== void 0 ? _f : false,
                output_audio_codec: (_g = data.output_audio_codec) !== null && _g !== void 0 ? _g : "mp3",
                output_audio_bitrate: (_h = data.output_audio_bitrate) !== null && _h !== void 0 ? _h : "128k",
                dict_id: data.dict_id,
                min_buffer_size: (_j = data.min_buffer_size) !== null && _j !== void 0 ? _j : 50,
                max_chunk_length: (_k = data.max_chunk_length) !== null && _k !== void 0 ? _k : 150,
            },
        };
        this.sendJson(message);
    }
    convert(text) {
        this.assertSocketIsOpen();
        const message = {
            type: "text",
            data: { text },
        };
        this.sendJson(message);
    }
    flush() {
        this.assertSocketIsOpen();
        const message = { type: "flush" };
        this.sendJson(message);
    }
    ping() {
        this.assertSocketIsOpen();
        const message = { type: "ping" };
        this.sendJson(message);
    }
    /** Connect to the websocket and register event handlers. */
    connect() {
        this.socket.reconnect();
        this.socket.addEventListener("open", this.handleOpen);
        this.socket.addEventListener("message", this.handleMessage);
        this.socket.addEventListener("close", this.handleClose);
        this.socket.addEventListener("error", this.handleError);
        return this;
    }
    /** Close the websocket and unregister event handlers. */
    close() {
        this.socket.close();
        this.handleClose({ code: 1000 });
        this.socket.removeEventListener("open", this.handleOpen);
        this.socket.removeEventListener("message", this.handleMessage);
        this.socket.removeEventListener("close", this.handleClose);
        this.socket.removeEventListener("error", this.handleError);
    }
    /** Returns a promise that resolves when the websocket is open. */
    waitForOpen() {
        return __awaiter(this, void 0, void 0, function* () {
            if (this.socket.readyState === core.ReconnectingWebSocket.OPEN) {
                return this.socket;
            }
            return new Promise((resolve, reject) => {
                this.socket.addEventListener("open", () => {
                    resolve(this.socket);
                });
                this.socket.addEventListener("error", (event) => {
                    reject(event);
                });
            });
        });
    }
    /** Asserts that the websocket is open. */
    assertSocketIsOpen() {
        if (!this.socket) {
            throw new Error("Socket is not connected.");
        }
        if (this.socket.readyState !== core.ReconnectingWebSocket.OPEN) {
            throw new Error("Socket is not open.");
        }
    }
    /** Send a binary payload to the websocket. */
    sendBinary(payload) {
        this.socket.send(payload);
    }
    /** Send a JSON payload to the websocket. */
    sendJson(payload) {
        const jsonPayload = toJson(payload);
        this.socket.send(jsonPayload);
    }
}
